From d9e8359b8a9df28f7bb0402db7835279c37d03c0 Mon Sep 17 00:00:00 2001
From: Alon Rotman <alon.rotman@solid-run.com>
Date: Wed, 21 Oct 2020 15:42:16 +0300
Subject: [PATCH 1/2] plat: marvell: octeontx2: SPD I2C bus

PCB revision 1.0 of SolidRun octeontx2 COM express type 7 has wrong
port.
This port will be fixed in revision 1.1 and this patch won't be needed
anymore.

In general revision 1.0 was sampled in small quanitites; so no backward
compatibility is required.

Signed-off-by: Alon Rotman <alon.rotman@solid-run.com>
---
 .../octeontx/otx2/t91/t9130/board/dram_port.c | 26 ++++++++++++-------
 1 file changed, 16 insertions(+), 10 deletions(-)

diff --git a/plat/marvell/octeontx/otx2/t91/t9130/board/dram_port.c b/plat/marvell/octeontx/otx2/t91/t9130/board/dram_port.c
index 0befadfc6..358d86b23 100644
--- a/plat/marvell/octeontx/otx2/t91/t9130/board/dram_port.c
+++ b/plat/marvell/octeontx/otx2/t91/t9130/board/dram_port.c
@@ -14,10 +14,16 @@
 #include <mvebu_def.h>
 #include <plat_marvell.h>
 
-#define MVEBU_CP_MPP_CTRL37_OFFS		20
-#define MVEBU_CP_MPP_CTRL38_OFFS		24
-#define MVEBU_CP_MPP_CTRL37_I2C0_SCK_ENA	0x2
-#define MVEBU_CP_MPP_CTRL38_I2C0_SDA_ENA	0x2
+//#define MVEBU_CP_MPP_CTRL37_OFFS		20
+//#define MVEBU_CP_MPP_CTRL38_OFFS		24
+//#define MVEBU_CP_MPP_CTRL37_I2C0_SCK_ENA	0x2
+//#define MVEBU_CP_MPP_CTRL38_I2C0_SDA_ENA	0x2
+
+
+#define MVEBU_CP_MPP_CTRL35_OFFS                12
+#define MVEBU_CP_MPP_CTRL36_OFFS                16
+#define MVEBU_CP_MPP_CTRL35_I2C1_SCK_ENA        0x2
+#define MVEBU_CP_MPP_CTRL36_I2C1_SDA_ENA        0x2
 
 #define MVEBU_MPP_CTRL_MASK			0xf
 
@@ -121,12 +127,12 @@ static void mpp_config(void)
 	reg = MVEBU_CP_MPP_REGS(0, 4);
 	/* configure CP0 MPP 37 and 38 to i2c */
 	val = mmio_read_32(reg);
-	val &= ~((MVEBU_MPP_CTRL_MASK << MVEBU_CP_MPP_CTRL37_OFFS) |
-		(MVEBU_MPP_CTRL_MASK << MVEBU_CP_MPP_CTRL38_OFFS));
-	val |= (MVEBU_CP_MPP_CTRL37_I2C0_SCK_ENA <<
-			MVEBU_CP_MPP_CTRL37_OFFS) |
-		(MVEBU_CP_MPP_CTRL38_I2C0_SDA_ENA <<
-			MVEBU_CP_MPP_CTRL38_OFFS);
+	val &= ~((MVEBU_MPP_CTRL_MASK << MVEBU_CP_MPP_CTRL35_OFFS) |
+		(MVEBU_MPP_CTRL_MASK << MVEBU_CP_MPP_CTRL36_OFFS));
+	val |= (MVEBU_CP_MPP_CTRL35_I2C1_SCK_ENA <<
+			MVEBU_CP_MPP_CTRL35_OFFS) |
+		(MVEBU_CP_MPP_CTRL36_I2C1_SDA_ENA <<
+			MVEBU_CP_MPP_CTRL36_OFFS);
 	mmio_write_32(reg, val);
 }
 
-- 
2.25.1

